torch/csrc/generic/StorageSharing.cpp

```diff
@@ -248,52 +292,153 @@ static PyObject * THPStorage_(shareCuda)(THPStorage *self)
   //     as key in shared_cache(multiprocessing/reduction.py).
   //     Offset in numel cannot uniquely represent a storage.
   PyTuple_SET_ITEM(tuple.get(), 3, _offset_bytes.release());
+  PyTuple_SET_ITEM(tuple.get(), 4, _ref_counter.release());
+  PyTuple_SET_ITEM(tuple.get(), 5, _ref_counter_offset.release());
+  PyTuple_SET_ITEM(tuple.get(), 6, _event_handle.release());
   return tuple.release();
   END_HANDLE_TH_ERRORS
 }
 
+static PyObject * THPStorage_(releaseIPCCounter)(PyObject *_unused, PyObject *args)
+{
+  HANDLE_TH_ERRORS
+  THPUtils_assert(PyTuple_GET_SIZE(args) == 2, "tuple of 2 items expected");
+  PyObject *_ref_counter = PyTuple_GET_ITEM(args, 0);
+  PyObject *_ref_counter_offset = PyTuple_GET_ITEM(args, 1);
+  if (!(_ref_counter != Py_None && PyBytes_Check(_ref_counter) &&
+        THPUtils_checkLong(_ref_counter_offset))) {
+    THPUtils_invalidArguments(
+        args,
+        nullptr,
+        "_release_ipc_counter in CUDA mode",
+        1,
+        "(bytes _ref_counter, int _ref_counter_offset)");
+    return nullptr;
+  }
+  std::string ref_counter_handle = PyBytes_AS_STRING(_ref_counter);
+  ptrdiff_t ref_counter_offset =
+      (ptrdiff_t)THPUtils_unpackLong(_ref_counter_offset);
+  // We don't want to break existing code, so resource deletion is best
+  // effort basis. Exception expected if producer process terminated
+  // before consumer released data.
+  int flags =
+      TH_ALLOCATOR_MAPPED_SHAREDMEM | TH_ALLOCATOR_MAPPED_NOCREATE;
+  try {
+    auto sptr = THRefcountedMapAllocator::makeDataPtr(
+        ref_counter_handle.c_str(),
+        flags,
+        sizeof(int64_t) * torch::CUDA_IPC_REF_COUNTER_FILE_SIZE,
+        nullptr);
+    *(static_cast<int64_t*>(sptr.get()) + ref_counter_offset) -= 1;
+  } catch (c10::Error) {
```

Comment:
Is it possible to make this catch more specific? I get very nervous about catch-all blocks like here, it's easy to mask other problems this way.

Reply:
Unfortunately all THMapAllocator-s throws errors with using AT_ERROR. So I have no way to distinguish file not found vs all other errors.

Reply:
I think it would be reasonable to adjust THMapAllocator to throw a subclass of Error and modify those sites? Just to double check: you added this try-catch because it solves a specific problem you saw in testing, is that right?

Reply:
Exactly. The specific problem is: crappy code with cuda ipc usage eventually ends producer process faster than consumer release mem handle. It leads to the cleanup of shared memory files (by producer) and multiple AT_ERRORS inside consumer when it tries to release counters. 

I already LOG(WARNING) in such cases and I want to avoid burring of this warning among multiple related (but unreadable) errors.